algorithms/HPdclassifier/R/hpdegbm.R

```diff
@@ -0,0 +1,288 @@
+       var.names = NULL,
+       #response.name = "y",
+       group = NULL,
+       trace = FALSE,  # If TRUE, hpdegbm will print out progress outside gbm.fit R function
+       completeModel = FALSE) # default system parameters are defined here
+
+# X_train: a data frame or data matrix containing the predictor variables
+# Y_train: a vector of outputs
+# dl_GBM_model: dlist storing the trained GBM model
+# dbest.iter: darray storing the best iterations of submodels
+# distribution: support Gaussian, AdaBoost, bernoulli, multinomial in DR version 1.2
+# n.trees: the total numner of tress to fit 
+# interactive.depth: the maximum depth of variable interactions
+# n.minobsinnode: minimum number of obervations in the tress terminal nodes
+# shrinkage: learning rate. Typical vaule: [0.001, 1]
+# bag.fraction: the fraction of the training set observations randomly selected for the next tree in the expansion
+# verbose: If TRUE, gbm prints out progress and performance indicators
+# trace: If TRUE, print out the training time
+# completeModel: If TRUE, add training time in the model 
+
+### start of hpdegbm function
+{
+   # parameters checking
+   if(missing(X_train))
+	stop("'X_train' is a required argument")
+
+   if(missing(Y_train))
+	stop("'Y_train' is a required argument")
+
+   if(!is.dframe(X_train) && !is.data.frame(X_train) && !is.darray(X_train) && !is.matrix(X_train))
+       stop("'X_train' must be a dframe or data.frame or darray or matrix")
+
+   if(!is.dframe(Y_train) && !is.data.frame(Y_train) && !is.darray(Y_train) && !is.matrix(Y_train) && !is.vector(Y_train))
+       stop("'Y_train' must be a dframe or data.frame or darray or matrix or numeric vector")
+
+   if(missing(dl_GBM_model))
+	stop("'dl_GBM_model' is a required argument")
+
+   if(missing(dbest.iter))
+	stop("'dbest.ier' is a required argument")
+
+   if(missing(nExecutor))
+	stop("'nExecutor' is a required argument")
+
+    nExecutor <- round(nExecutor)
+    if(nExecutor <= 0)
+        stop("nExecutor should be a positive integer number")
+
+   if(missing(distribution))
+	stop("'distribution' is a required argument")
+
+   if(n.trees <= 0)
+        stop("'n.trees' must be more than 0")
+
+   if(interaction.depth <= 0)
+        stop("'interaction.depth' must be more than 0")
+
+
+   if ((interaction.depth%%1 < 0) && ((interaction.depth%%1 > 0)))
+        stop("'interaction.depth' must be an integer")
+
+   if(n.minobsinnode <= 0)
+        stop("'n.minobsinnode' must be more than 0")
+
+   if ((n.minobsinnode%%1 < 0) && ((n.minobsinnode%%1 > 0)))
+        stop("'n.minobsinnode' must be an integer")
+
+   if(shrinkage <= 0)
+        stop("'shrinkage' must be more than 0")
+
+   if ((shrinkage < 0.001) && ((shrinkage > 1)))
+        stop("'shrinkage' must be between [0.001,1]")
+
+
+   if(bag.fraction <= 0)
+        stop("'bag.fraction' must be more than 0")
+
+   if (bag.fraction > 1)
+        stop("'shrinkage' must be (0,1]")
```

Comment:
Just as a good programming practice, it would better if you combine the two condition for the range of "bag.fraction" using OR, and print out a single error message that informs about the valid range.